contrib/core-contract-tests/tests/pox-4/pox_StackAggregationCommitAuthCommand.ts

```diff
@@ -0,0 +1,148 @@
+ *
+ * This command calls stack-aggregation-commit using an `authorization`.
+ *
+ * Constraints for running this command include:
+ * - The Operator must have locked STX on behalf of at least one stacker.
+ * - The total amount previously locked by the Operator on behalf of the
+ *   stackers has to be greater than the uSTX threshold.
+ * - All of the Stackers must have delegated to the same pox address.
+ */
+export class StackAggregationCommitAuthCommand implements PoxCommand {
+  readonly operator: Wallet;
+  readonly authId: number;
+  readonly currentCycle: number;
+
+  /**
+   * Constructs a `StackAggregationCommitAuthCommand` to lock uSTX for stacking.
+   *
+   * @param operator - Represents the `Operator`'s wallet.
+   * @param authId - Unique `auth-id` for the authorization.
+   * @param currentCycle - The current reward cycle.
+   */
+  constructor(
+    operator: Wallet,
+    authId: number,
+    currentCycle: number,
+  ) {
+    this.operator = operator;
+    this.authId = authId;
+    this.currentCycle = currentCycle;
+  }
+
+  check(model: Readonly<Stub>): boolean {
+    // Constraints for running this command include:
+    // - The Operator must have locked STX on behalf of at least one stacker.
+    // - The total amount previously locked by the Operator on behalf of the
+    //   stackers has to be greater than the uSTX threshold.
+    // - All of the Stackers must have delegated to the same pox address.
+    let sameDelegatedPoxAddrAllStackers = true;
+    const firstDelegatedPoxAddress = this.operator.lockedAddresses.length > 0
+      ? model.wallets.get(this.operator.lockedAddresses[0])!.delegatedPoxAddress
+      : this.operator.btcAddress;
+
+    this.operator.lockedAddresses.forEach((stacker) => {
+      if (
+        model.wallets.get(stacker)!.delegatedPoxAddress !==
+          firstDelegatedPoxAddress
+      ) sameDelegatedPoxAddrAllStackers = false;
+    });
+
+    return this.operator.lockedAddresses.length > 0 &&
+      this.operator.amountToCommit >= model.stackingMinimum &&
+      sameDelegatedPoxAddrAllStackers;
+  }
+
+  run(model: Stub, real: Real): void {
+    model.trackCommandRun(this.constructor.name);
+
+    const committedAmount = this.operator.amountToCommit;
+
+    const { result: setSignature } = real.network.callPublicFn(
+      "ST000000000000000000002AMW42H.pox-4",
+      "set-signer-key-authorization",
+      [
+        // (pox-addr (tuple (version (buff 1)) (hashbytes (buff 32))))
+        poxAddressToTuple(this.operator.btcAddress),
+        // (period uint)
+        Cl.uint(1),
+        // (reward-cycle uint)
+        Cl.uint(this.currentCycle + 1),
+        // (topic (string-ascii 14))
+        Cl.stringAscii("agg-commit"),
+        // (signer-key (buff 33))
+        Cl.bufferFromHex(this.operator.signerPubKey),
+        // (allowed bool)
+        Cl.bool(true),
+        // (max-amount uint)
+        Cl.uint(this.operator.amountToCommit),
```

Comment:
Any reason not to use the 'committedAmount' var set few lines above to this value?